project_from_quotation/models/sale_order.py

```diff
@@ -0,0 +1,231 @@
+                    .get_param("project_from_quotation.template_project_id")
+                )
+            )
+        except ValueError:
+            _logger.warning(
+                "Project template ID is not set or invalid. "
+                "Make sure you have configured it in settings."
+            )
+
+    @api.model
+    def _get_quotation_task_template(self):
+        """
+        Method to obtain a template for quotation task
+        from settings. If not set or is invalid, logs a warning
+
+        Returns:
+            project.task record or None
+        """
+        try:
+            return self.env["project.task"].browse(
+                int(
+                    self.env["ir.config_parameter"]
+                    .sudo()
+                    .get_param("project_from_quotation.template_task_id")
+                )
+            )
+        except ValueError:
+            _logger.warning(
+                "Task template ID is invalid or not set. "
+                "Make sure you have configured it in settings."
+            )
+
+    def _get_order_lines_unique_products(self, order_lines):
+        """
+        Get order lines with unique products
+
+        Args:
+            order_lines (sale.order.line recordset): Order lines
+            if not set, uses self.order_line
+
+        Returns:
+            sale.order.line recordset: Every first
+            line with the same product
+        """
+        self.ensure_one()
+        unique_products = []
+        res_order_lines = self.env["sale.order.line"]
+        order_lines = order_lines.filtered("product_id")
+
+        for line in order_lines:
+            if line.product_id.id not in unique_products:
+                unique_products.append(line.product_id.id)
+                res_order_lines |= line
+
+        return res_order_lines
+
+    def _create_project_from_quotation(
+        self,
+        order_lines,
+        mode="project",
+    ):
+        """
+        Create project from a quotation
+
+        Args:
+        order_lines (sale.order.line): Order lines to use for tasks creation
+        mode (str): Project creation mode. Possible values:
+            project - create a new empty project
+            tasks_per_line - create a project with tasks for each order line
+            tasks_per_product - create a project with tasks for each product
+
+        Returns:
+            Created record of the project.project model
+        """
+        self.ensure_one()
+        project = False
+        task_obj = self.env["project.task"]
+        project_obj = self.env["project.project"]
+        order_lines = order_lines.filtered(lambda x: not x.task_id)
+
+        project_template = self._get_quotation_project_template()
+        if project_template:
+            project = project_template.copy(self._prepare_project_vals(self))
+        else:
+            project = project_obj.create(self._prepare_project_vals(self))
+
+        if mode.startswith("tasks"):
```

Comment:
I would have avoided to use such things. Prefer: 
```suggestion
        if mode in ["tasks_per_line", "tasks_per_product"]:
```

Reply:
Hello @rousseldenis , could you please check the recent updates? 
We have applied changes you requested.